ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/command/toolbar/controller/ToolbarControllerViewImpl.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.command.toolbar.controller;
+
+import com.google.gwt.core.client.JavaScriptObject;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+/**
+ * Top menu widget implementing ToolbarControllerView.
+ */
+@Singleton
+public class ToolbarControllerViewImpl extends Composite implements ToolbarControllerView {
+
+    interface ToolbarControllerViewImplUiBinder extends UiBinder<Widget, ToolbarControllerViewImpl> {
+    }
+
+    private ActionDelegate delegate;
+
+    @Inject
+    public ToolbarControllerViewImpl(ToolbarControllerViewImplUiBinder uiBinder) {
+        initWidget(uiBinder.createAndBindUi(this));
+
+        addDomHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent clickEvent) {
+                int left = getAbsoluteLeft(getWidget().getElement());
+                int top = getAbsoluteTop(getWidget().getElement());
+                delegate.showMenu(left + 20, top + 35);
+            }
+        }, ClickEvent.getType());
+    }
+
+    /**
+     * Returns absolute left position of the element.
+     *
+     * @param element
+     *          element
+     * @return
+     *          element left position
+     */
+    private native int getAbsoluteLeft(JavaScriptObject element) /*-{
```

Comment:
It would be better to move this method to `org.eclipse.che.ide.util.dom.DomUtils`